src/Api/HL/Middleware/InternalAuthMiddleware.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2023 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Api\HL\Middleware;
+
+/**
+ * This middleware is not loaded by default when using the API.
+ * It may be manually added on the Router instance in cases where GLPI itself needs to access the API.
+ * If a user is already logged in, this middleware will allow the request to continue.
+ */
+class InternalAuthMiddleware extends AbstractMiddleware implements AuthMiddlewareInterface
+{
+    public function process(MiddlewareInput $input, callable $next): void
+    {
+        if (\Session::getLoginUserID()) {
```

Comment:
Auth middleware handling is a bit confusing. By default, the router prepares a 401 error response. If a middleware removes that response, the router treats is as a successful authentication. If there is a logged in user, this middleware removes that response and doesn't call the next middleware (no need to). Otherwise, it goes to the next middleware one if it exists.

After all auth middleware runs, if there is still a 401 response ready, it tries the built-in authentication which is now OAuth.